roachprod-microbench: standardize naming

herkolategan opened the pull request:
Previously, we would refer to "new" and "old" as the revisions microbenchmarks
ran against. This has been updated to use a more descriptive naming scheme.

Baseline, previously "old", refers to the base revision compared against, which
will usually be a stable release. Experiment, previously "new", refers to the
experimental revision, a PR, or merge to master which is being compared against
the stable baseline.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/128184)
<!-- Reviewable:end -->

herkolategan committed:
roachprod-microbench: standardize naming

Previously, we would refer to "new" and "old" as the revisions microbenchmarks
ran against. This has been updated to use a more descriptive naming scheme.

Baseline, previously "old", refers to the base revision compared against, which
will usually be a stable release. Experiment, previously "new", refers to the
experimental revision, a PR, or merge to master which is being compared against
the stable baseline.

Epic: None
Release note: None

herkolategan commented on pkg/cmd/roachprod-microbench/testdata/compare:
```diff
@@ -3,34 +3,34 @@ compare name-conflict-a name-conflict-b
 ----
 Package pkg/parent
 Metric B/op
-BenchmarkEntry pkg/parent/sub→WithNameConflict ~ [2200 2000] p=1.000 n=1
```
Note: The test data is expected to change as it is ordered by the "name" to be deterministic, which has now changed alphabetic order "new", "old" -> "baseline", "experiment" (where "new" use to be first).

herkolategan commented:
TFTR!

bors r=srosenberg

herkolategan commented:
bors retry
